crates/qasm3/src/exporter.rs

```diff
@@ -0,0 +1,1412 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::ast::{
+    Alias, Barrier, BitArray, Break, ClassicalDeclaration, ClassicalType, Continue, Delay,
+    Designator, DurationLiteral, DurationUnit, Expression, Float, GateCall, Header, IODeclaration,
+    IOModifier, Identifier, IdentifierOrSubscripted, Include, IndexSet, IntegerLiteral, Node,
+    Parameter, Program, QuantumBlock, QuantumDeclaration, QuantumGateDefinition,
+    QuantumGateSignature, QuantumInstruction, QuantumMeasurement, QuantumMeasurementAssignment,
+    Reset, Statement, SubscriptedIdentifier, Version,
+};
+use std::io::Write;
+
+use crate::printer::BasicPrinter;
+use hashbrown::{HashMap, HashSet};
+use indexmap::IndexMap;
+use pyo3::prelude::*;
+use pyo3::Python;
+use qiskit_circuit::bit::{
+    ClassicalRegister, QuantumRegister, Register, ShareableClbit, ShareableQubit,
+};
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::operations::{DelayUnit, StandardInstruction};
+use qiskit_circuit::operations::{Operation, Param};
+use qiskit_circuit::packed_instruction::PackedInstruction;
+use thiserror::Error;
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+type ExporterResult<T> = Result<T, QASM3ExporterError>;
+
+// These are the prefixes used for the loose qubit and bit names.
+lazy_static! {
+    static ref BIT_PREFIX: &'static str = "_bit";
+}
+lazy_static! {
+    static ref QUBIT_PREFIX: &'static str = "_qubit";
+}
+
+// These are the prefixes used for the gate parameters and qubits.
+lazy_static! {
+    static ref GATE_PARAM_PREFIX: &'static str = "_gate_p";
+}
+lazy_static! {
+    static ref GATE_QUBIT_PREFIX: &'static str = "_gate_q";
+}
```

Comment:
These ones don't need to be lazy - `const BIT_PREFIX: &str = "_bit";` would work fine.